kong/vault_service_test.go

```diff
@@ -0,0 +1,202 @@
+
+	_, err = client.Vaults.Update(defaultCtx, nil)
+	require.EqualError(err, "cannot update a nil vault")
+}
+
+func TestVaultWithTags(t *testing.T) {
+	RunWhenEnterprise(t, ">=3.0.0", RequiredFeatures{})
+	require := require.New(t)
+
+	client, err := NewTestClient(nil, nil)
+	require.NoError(err)
+	require.NotNil(client)
+
+	vault := &Vault{
+		Name:        String("env"),
+		Prefix:      String("my-env-vault"),
+		Description: String("ENV vault for secrets"),
+		Config: Configuration{
+			"prefix": "SECURE_",
+		},
+		Tags: StringSlice("tag1", "tag2"),
+	}
+
+	createdVault, err := client.Vaults.Create(defaultCtx, vault)
+	require.NoError(err)
+	require.NotNil(createdVault)
+	require.Equal(StringSlice("tag1", "tag2"), createdVault.Tags)
+
+	err = client.Vaults.Delete(defaultCtx, createdVault.ID)
+	require.NoError(err)
+}
+
+func TestVaultListEndpoint(t *testing.T) {
+	RunWhenEnterprise(t, ">=3.0.0", RequiredFeatures{})
+	require := require.New(t)
+
+	client, err := NewTestClient(nil, nil)
+	require.NoError(err)
+	require.NotNil(client)
+
+	// fixtures
+	vaults := []*Vault{
+		{
+			Name:        String("env"),
+			Prefix:      String("my-env-vault"),
+			Description: String("ENV vault for secrets"),
+			Config: Configuration{
+				"prefix": "SECURE_",
+			},
+		},
+		{
+			Name:        String("aws"),
+			Prefix:      String("my-aws-vault"),
+			Description: String("aws vault for secrets"),
+			Config: Configuration{
+				"region": "us-east-2",
+			},
+		},
+		{
+			Name:        String("hcv"),
+			Prefix:      String("my-hcv-vault"),
+			Description: String("hcv vault for secrets"),
+			Config: Configuration{
+				"token": "super-secret-token",
+			},
+		},
+	}
+
+	// create fixturs
+	for i := 0; i < len(vaults); i++ {
+		vault, err := client.Vaults.Create(defaultCtx, vaults[i])
+		require.NoError(err)
+		require.NotNil(vault)
+		vaults[i] = vault
+	}
+
+	vaultsFromKong, next, err := client.Vaults.List(defaultCtx, nil)
+	require.NoError(err)
+	require.Nil(next)
+	require.NotNil(vaultsFromKong)
+	require.Equal(3, len(vaultsFromKong))
+
+	// check if we see all vaults
+	require.True(compareVaults(t, vaults, vaultsFromKong))
+
+	// Test pagination
+	vaultsFromKong = []*Vault{}
+
+	// first page
+	page1, next, err := client.Vaults.List(defaultCtx, &ListOpt{Size: 1})
+	require.NoError(err)
+	require.NotNil(next)
+	require.NotNil(page1)
+	require.Equal(1, len(page1))
+	vaultsFromKong = append(vaultsFromKong, page1...)
+
+	// last page
+	next.Size = 2
+	page2, next, err := client.Vaults.List(defaultCtx, next)
+	require.NoError(err)
+	require.Nil(next)
+	require.NotNil(page2)
+	require.Equal(2, len(page2))
+	vaultsFromKong = append(vaultsFromKong, page2...)
+
+	require.True(compareVaults(t, vaults, vaultsFromKong))
+
+	vaults, err = client.Vaults.ListAll(defaultCtx)
+	require.NoError(err)
+	require.NotNil(vaults)
+	require.Equal(3, len(vaults))
+
+	for i := 0; i < len(vaults); i++ {
+		require.NoError(client.Vaults.Delete(defaultCtx, vaults[i].ID))
```

Comment:
Oh, wait. Is that meant to clean up the environment? If that's the case I suggest to use something like

```go
t.Cleanup(func(){
  // Note the assert here as we might want more logic to be run as part of the cleanup,
  // regardless of the return value of removing this particular object.
  assert.NoError(client.Vaults.Delete(defaultCtx, vaults[i].ID))
})
```

called just after a successful `.Create()`.